### PR TITLE
Fixes #9095: Use a 2 steps download in rudder agent makefile

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/9095

### CODE REVIEW

File: rudder-agent/SOURCES/Makefile [hunk 1]
```diff
@@ -37,24 +37,31 @@ INSTALL_DIR = $(DESTDIR)$(RUDDER_DIR)
 TRY_LDFLAGS=-pie -Wl,-z,relro -Wl,-z,now
 TRY_CFLAGS=-fPIE -fstack-protector
 
-# Autodetect wget or curl or curl usage and proxy configuration
+# GENERIC_GET: do note remove this line
```
Comment (jooooooon) [inline rudder-agent/SOURCES/Makefile]: s/note/not

File: rudder-agent/SOURCES/Makefile [hunk 2]
```diff
@@ -37,24 +37,31 @@ INSTALL_DIR = $(DESTDIR)$(RUDDER_DIR)
...
+# This is a generic get code that is common to many of our Makefiles.
+# The tag is here to find them.
+# If you change this code, change all the places where you can find this tag too.
+# It is duplicated because it is the code that is used to manage dependency
+# to other repositories, hence allowing deduplication of other code.
+# Autodetect wget or curl usage and proxy configuration
```
Comment (jooooooon) [inline rudder-agent/SOURCES/Makefile]: or fetch

File: rudder-agent/SOURCES/Makefile [hunk 3]
```diff
@@ -37,24 +37,31 @@ INSTALL_DIR = $(DESTDIR)$(RUDDER_DIR)
...
-GET = $(PROXY_ENV) $(FETCH)
+_GET = $(PROXY_ENV) $(FETCH)
 else
-GET = $(PROXY_ENV) $(WGET)
+_GET = $(PROXY_ENV) $(WGET)
 GET_OPT = --wget
 CPAN_OPT = --wget
 endif
 endif
+GET=get() { $(_GET) "$$1.part" "$$2" && mv "$$1.part" "$$1"; }; get
```
Comment (jooooooon) [inline rudder-agent/SOURCES/Makefile]: Could you add an example of how this will be called to the "doc" comment above?

**Comment by peckpeck** [comment]
Commit modified

**Comment by Kegeruneku** [comment]
looks good now

**Pull request merged**